Automattic/jetpack
# E2E tests: fix plan data consistency

Plan data is not consistent right after the connection. It was relatively fine with the purchase flow, but now when we switched to mocked plan data, the problem occurs again. Symptoms are the following:

1. Simple payment and/or WordAds blocks are not available in the block editor, while the Jetpack dashboard shows Pro plan.
2. Frontends 'Initial_State' plan is different from what is set on backend.
3. WordAds module could get deactivated after it was successfully activated.

Seems like 1 and 3 is just a result of 2. 3 might be related to this logic which deactivates the module if "current" plan does not support it: https://github.com/Automattic/jetpack/blob/master/_inc/lib/admin-pages/class.jetpack-admin-page.php#L175

To prevent this, I trying the following:

1. Sync frontend and backend plan data by reloading dashboard page
2. activate the required modules after that. 

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.
1156386383419466-as-1158682698175497

#### Testing instructions:






* Check that CI E2E tests are green
* Maybe restart the build, and check if they passing

#### Proposed changelog entry for your changes:

* n/a
